examples/next-js-netlify/Earthfile

```
@@ -0,0 +1,38 @@
+VERSION 0.6
+FROM --platform=linux/amd64 node:latest
+WORKDIR /app
+
+deps:
+    ARG auth_token
+    ARG site_id
+    # Set environment variables required for building and deploying to Netlify
+    ENV NETLIFY_AUTH_TOKEN=$auth_token
+    ENV NETLIFY_SITE_ID=$site_id
+    # Install the Netlify CLI (global)
+    RUN npm install netlify-cli -g
+    # Copy package.json for installing required packages
+    COPY package.json ./
+    # Install the Netlify CLI (local) and Netlify Next.js plugin and required packages
+    RUN npm install netlify-cli --save-dev
+    RUN npm install @netlify/plugin-nextjs --save-dev
+    RUN npm install
+    # Copy netlify.toml - required for building and deploying to Netlify
+    COPY netlify.toml ./
+
+build:
+    FROM +deps
+    # Copy files and directories required for building
+    COPY next-env.d.ts styles.module.css tsconfig.json ./
+    COPY --dir pages ./
+    # Build site
+    RUN netlify build --context production
+    SAVE ARTIFACT ./node_modules node_modules/ AS LOCAL ./
+    SAVE ARTIFACT ./.next .next/ AS LOCAL ./
+    SAVE ARTIFACT ./.netlify .netlify/ AS LOCAL ./
+
+deploy:
+    FROM +deps
+    # Copy artifacts required for deploying to Netlify
+    COPY +build/node_modules/ +build/.next/ +build/.netlify/ ./
+    # Deploy site
+    RUN netlify deploy --prod
```

We generally use `--push` commands for things that have effects outside of Earthly.

`RUN --push` commands ensure that the execution takes place every time, even if it is cached. It is also a safety feature - `RUN --push` commands execute only when `earthly` itself is executed with the `--push` flag: `earthly --push +deploy`. A more in-depth explanation here: https://docs.earthly.dev/best-practices#use-run-push-for-deployment-commands.

```suggestion
    RUN --push netlify deploy --prod
```

---

I didn't know that. I though push was more for pushing container images. I'll update.